src/DataTable/Pagination.tsx

```diff
@@ -202,7 +238,7 @@ export function Pagination({
             </Button>
           </Step>
           {pageCount > 0 ? (
-            <Step>
+            <Step hiddenViewportRanges={getViewportRangesToHidePages()}>
```

Comment:
Is the `hiddenViewportRanges` prop something that can be placed on `TablePaginationSteps`? Would be great to calculate this once and place it on the outermost node and then target it e.g.

```css
  @media ${viewportRanges.narrow} {
    .TablePaginationSteps[data-hidden-viewport-ranges*='narrow'] > *:not(:first-child):not(:last-child) {
      display: none;
    }

    .TablePaginationSteps[data-hidden-viewport-ranges*='narrow'] > *:first-child {
      margin-inline-end: 0;
    }
  }
```

If not, I think this is something we should calculate once instead of as a function and just pass its value to each `Step` instead of messing around with `useCallback()` and calling it for each step.

Reply:
Ahh good catch! Ok, I'll refactor 👍